ledger/persistedonlineaccts_list.go

```diff
@@ -0,0 +1,144 @@
+// Copyright (C) 2019-2022 Algorand, Inc.
+// This file is part of go-algorand
+//
+// go-algorand is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// go-algorand is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with go-algorand.  If not, see <https://www.gnu.org/licenses/>.
+
+package ledger
+
+// persistedOnlineAccountDataList represents a doubly linked list.
+// must initiate with newPersistedAccountList.
+type persistedOnlineAccountDataList struct {
```

Comment:
With Go 1.18 on the horizon, can we consider the native library's [double linked list implementation](https://pkg.go.dev/container/list)?

Reply:
There is already persistedaccts_list.go, persistedresources_list.go, and now persistedonlineaccts_list.go and they are all identical except for the value type. So this seems like a good candidate to collapse these into a single implementation as soon as we upgrade to Go 1.18.

Reply:
If I recall right we use concrete types instead of list.Node because of performance penalty on large lists. These will be fixed to generics/templates.